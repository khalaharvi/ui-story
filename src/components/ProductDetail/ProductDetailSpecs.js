import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { cx } from './ProductDetail';

const ProductDetailSpecs = ({ specs }) => (
  <div className={classnames(cx('specs'), 'columns')}>
    {
      specs.map(spec => (
        <div className="col" key={spec.name}>
          <h2 className={cx('title')}>{spec.name}</h2>
          <dl>
            {
              spec.items.map(item => (
                <Fragment key={item.name}>
                  <dt>{item.name}</dt>
                  <dd>{item.value}</dd>
                </Fragment>
              ))
            }
          </dl>
        </div>
      ))
    }
  </div>
);

ProductDetailSpecs.propTypes = {
  specs: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    items: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })).isRequired
  })).isRequired
};

export default ProductDetailSpecs;
