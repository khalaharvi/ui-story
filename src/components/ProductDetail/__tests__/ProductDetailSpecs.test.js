import React from 'react';
import { shallow } from 'enzyme';

import ProductDetailSpecs from '../ProductDetailSpecs';

describe('<ProductDetailSpecs />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProductDetailSpecs
        specs={[
          {
            name: 'Product Specifications',
            items: [
              {
                name: 'Dimensions',
                value: 'Height: 26.5 in. Diameter: 11″ Base: 13″h'
              },
              {
                name: 'Materials',
                value: 'Bluz'
              }
            ]
          },
          {
            name: 'Vintage Specifications',
            items: [
              {
                name: 'Period',
                value: '1940-1949'
              },
              {
                name: 'Creator',
                value: 'Unknown'
              },
              {
                name: 'Condition',
                value: 'Great'
              }
            ]
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
