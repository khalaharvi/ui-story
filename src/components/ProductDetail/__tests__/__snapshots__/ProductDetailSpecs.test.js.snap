// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailSpecs /> renders correctly 1`] = `
<div
  className="dh-product-detail-specs columns"
>
  <div
    className="col"
    key="Product Specifications"
  >
    <h2
      className="dh-product-detail-title"
    >
      Product Specifications
    </h2>
    <dl>
      <React.Fragment
        key="Dimensions"
      >
        <dt>
          Dimensions
        </dt>
        <dd>
          Height: 26.5 in. Diameter: 11″ Base: 13″h
        </dd>
      </React.Fragment>
      <React.Fragment
        key="Materials"
      >
        <dt>
          Materials
        </dt>
        <dd>
          Bluz
        </dd>
      </React.Fragment>
    </dl>
  </div>
  <div
    className="col"
    key="Vintage Specifications"
  >
    <h2
      className="dh-product-detail-title"
    >
      Vintage Specifications
    </h2>
    <dl>
      <React.Fragment
        key="Period"
      >
        <dt>
          Period
        </dt>
        <dd>
          1940-1949
        </dd>
      </React.Fragment>
      <React.Fragment
        key="Creator"
      >
        <dt>
          Creator
        </dt>
        <dd>
          Unknown
        </dd>
      </React.Fragment>
      <React.Fragment
        key="Condition"
      >
        <dt>
          Condition
        </dt>
        <dd>
          Great
        </dd>
      </React.Fragment>
    </dl>
  </div>
</div>
`;
